### ISSUE TITLE
Need export and import capabilities for templates

### ISSUE THREAD

**Issue by vikasrohit** [topcoder-platform/tc-project-service#170]
It is very tedious to create templates in each environment from scratch. Purpose of this task is to create two endpoints, one to export and another to import all existing templates, for `projectTemplate` and `productTemplate` models.

Exact requirements on the import behaviour needed to be detailed which would be more suitable after we are done with #109.

Open questions:
1. Should we import by id or key?
2. Should we override or skip, while import, existing one when there exists a template with same id/key
3. Should we always  insert a new record while importing? May be with versioning support we can do inserts for all imported templates while importing if import is for new version and override existing ones if import is for existing version.

**Comment by gondzo** [comment]
This is similar to the local test data issue -> https://github.com/topcoder-platform/tc-project-service/issues/102

Importing by id would be easier to maintain in the future, but it will be tricky if existing templates don't have same ids in dev and prod. Override seems more natural when importing existing template. I guess all template changes will first be made in dev and then pushed to prod so there isn't really a case when you want to skip existing templates

**Comment by vikasrohit** [comment]
> I guess all template changes will first be made in dev and then pushed to prod so there isn't really a case when you want to skip existing templates

That is true but there might be cases where we have changes in templates in dev env which are to be pushed in multiple releases to production. In that case, we would be left with moving all changes together to the production.

Also, we already have some differences in ids of the templates now between dev and production. So, I guess it would be difficult for us to sync their ids and I think that we should not do any way. IMO, we need the capability of configuring the behaviour of import while importing templates.

**Comment by gondzo** [comment]
@vikasrohit is this still needed?

**Comment by vikasrohit** [comment]
It might be needed in future. I would like to keep it open till our metadata management UI is mature enough to support all use cases needed by support team to manage templates in Connect.
